NoahPBM/contracts/paymentmanager/NoahPaymentManager.sol

```
@@ -325,16 +319,15 @@ contract NoahPaymentManager is Ownable, Pausable, AccessControl, INoahPaymentSta
         require(Address.isContract(campaignPBM), "Must be from a smart contract");
         require(Address.isContract(erc20Token), "Must be a valid ERC20 smart contract");
         require(erc20TokenValue > 0, "Token value should be more than 0");
-
         // Ensure that only campaign PBM can spend its own money
-        if (pbmTokenBalance[campaignPBM][erc20Token] >= erc20TokenValue) {
-            // Subtract from main balance directly to be sent out immediately.
-            _decreaseTreasuryBalance(campaignPBM, erc20Token, erc20TokenValue);
+        require(pbmTokenBalance[campaignPBM][erc20Token] >= erc20TokenValue, "ERC20: transfer amount exceeds balance");
```

noted, now it wont fail silently due to if condition not executing.